src/leader_schedule.rs

```diff
@@ -0,0 +1,46 @@
+use crate::leader_scheduler::LeaderScheduler;
+
+use solana_sdk::pubkey::Pubkey;
+
+pub struct LeaderSchedule {
+    slot_leaders: Vec<Pubkey>,
+    ticks_per_slot: u64,
+    ticks_per_epoch: u64,
```

Comment:
No redundant data please. Instead:

```rust
pub fn duration_in_ticks(&self) {
   self.ticks_per_slot * self.slot_leaders.len()
}
```

Reply:
That's not necessarily accurate. If you have 10 slots in an epoch, but only two leaders, slot_leaders will be length 2, and they just round robin through the epoch.

Reply:
Good point!